diff --git a/libcutils/fs_config.cpp b/libcutils/fs_config.cpp
index cb15b57..7face88 100644
--- a/libcutils/fs_config.cpp
+++ b/libcutils/fs_config.cpp
@@ -182,8 +182,8 @@ static const struct fs_path_config android_files[] = {
                                               "system/bin/run-as" },
 
     // Support FIFO scheduling mode in SurfaceFlinger.
-    { 00755, AID_SYSTEM,    AID_GRAPHICS,  CAP_MASK_LONG(CAP_SYS_NICE),
-                                              "system/bin/surfaceflinger" },
+//    { 00755, AID_SYSTEM,    AID_GRAPHICS,  CAP_MASK_LONG(CAP_SYS_NICE),
+  //                                            "system/bin/surfaceflinger" },
 
     // Support hostapd administering a network interface.
     { 00755, AID_WIFI,      AID_WIFI,      CAP_MASK_LONG(CAP_NET_ADMIN) |